### PR TITLE
Fix: Allow null data points for time series

### PR DESCRIPTION
**Pull request by kevinwcyu**
Currently, if a column includes [`TimeSeriesMeasureValueColumnInfo`](https://docs.aws.amazon.com/timestream/latest/developerguide/API_query_Type.html) we assume that it also has a time series value https://github.com/grafana/timestream-datasource/blob/7f68a0ba80cfdc8708989e121314266af5db5773/pkg/timestream/helper.go#L39-L53

However, it could just be a [null data point](https://docs.aws.amazon.com/timestream/latest/developerguide/API_query_Datum.html#timestream-Type-query_Datum-NullValue).

This PR allows data points to be a null value instead of assuming it must be present.

This fixes https://github.com/grafana/timestream-datasource/issues/168. You can see a screenshot of what the error looked like in the panel in the linked issue.

Currently this is what the Timestream query editor shows when there are null data points.
AWS timestream query editor with null data points

With the changes in this PR this is what Grafana looks like now
Grafana timestream panel with null data points

### CODE REVIEW

**Review by sunker — APPROVED** [review]
Nice work!

File: pkg/timestream/helper.go [hunk 1]
```diff
@@ -47,8 +47,9 @@ func QueryResultToDataFrame(res *timestreamquery.QueryOutput) (dr backend.DataRe
 		for _, timeseriesColumn := range timeseriesColumns {
 			for _, series := range res.Rows {
 				tv := series.Data[timeseriesColumn.columnIdx].TimeSeriesValue
-				if tv == nil {
-					dr.Error = fmt.Errorf("Expecting timeseries colum at: %d", timeseriesColumn.columnIdx)
+				isNullDataPoint := series.Data[timeseriesColumn.columnIdx].NullValue
+				if tv == nil && !*isNullDataPoint {
```
Comment (andresmgot) [inline pkg/timestream/helper.go:51]: you need to check if the pointer is null to avoid a possible panic

**Review by andresmgot — APPROVED** [review]
nice, I see you figured out how these integration tests work :)

File: pkg/timestream/helper.go (hunk 1 above)
Comment (kevinwcyu) [reply pkg/timestream/helper.go]: oof, nice catch.

**Comment by brunoleite** [comment]
Grafana shows "No Data" for this scenario when using influx db instead of an empty graph. I tried to replicate the same behavior with timestream, but I did not manage to do it.

**Comment by kevinwcyu** [comment]
> Grafana shows "No Data" for this scenario when using influx db instead of an empty graph. I tried to replicate the same behavior with timestream, but I did not manage to do it.

@brunoleite, if you use the following

```sql
SELECT interpolatedValue
FROM interpolatedTimeseries
WHERE interpolatedValue IS NOT NULL
```

instead of

```sql
SELECT *
FROM interpolatedTimeseries
```

It should not return any rows and will show "No Data".

**Comment by brunoleite** [comment]
Sorry for the delay. I have tried that (without this fix), and if I add `WHERE interpolatedValue IS NOT NULL` I only see a black graph, without the red badge.
Screen Shot 2022-06-17 at 1 18 43 PM

**Pull request merged**